### ISSUE TITLE
Linux: Ubuntu 16.04 installs GStreamer 1.0 series, not 0.10 series, by default

### ISSUE THREAD

**Issue by khaeru** [toggl/toggldesktop#1894]
In Ubuntu 16.04 ("xenial"), the core desktop applications depend on `libgstreamer1.0-0`, rather than `libgstreamer0.10-0`. The latter is not automatically installed, leading to the following error when running `TogglDesktop.sh`:
```
/home/khaeru/toggldesktop/./TogglDesktop: error while loading shared libraries: libgstapp-0.10.so.0: cannot open shared object file: No such file or directory
```

I was able to work around this by manually installing `libgstreamer-plugins-base0.10-0`, which contains the required library.

**Comment by sebastienserre** [comment]
Hello Khaeru,

Where did you find libgstreamer-plugins-base0.10-0 to make work Toggl ?

Thx

**Comment by chengguangnan** [comment]
Also, the deb install is not responding. I think for the same reason, failing silently in the backend. I have to download the tar ball package run ./TogglDesktop to get into the problem above.

**Comment by joshfleming** [comment]
This workaround worked for me. 
sudo apt-get install libgstreamer0.10-dev libgstreamer-plugins-base0.10-dev

**Comment by semente** [comment]
On Debian, there is no libgstreamer0.10 anymore.

**Comment by IndrekV** [comment]
`libgstreamer0.10` is a dependency of Qt 5.5 that we are currently using for the app. To make i work with newer gstreamer we need to probably update Qt and convert our app to work with the newer version.

**Comment by mflerackers** [comment]
Ubuntu 16.10 no longer provides libgstreamer0.10 packages

**Comment by Spurlos** [comment]
Guess it the right time to update the app to use latest QT.

**Comment by Spurlos** [comment]
And a quick workaround, tested on my machine with Ubuntu 16.10:

```
wget http://fr.archive.ubuntu.com/ubuntu/pool/main/g/gst-plugins-base0.10/libgstreamer-plugins-base0.10-0_0.10.36-1_amd64.deb
wget http://fr.archive.ubuntu.com/ubuntu/pool/universe/g/gstreamer0.10/libgstreamer0.10-0_0.10.36-1.5ubuntu1_amd64.deb
sudo dpkg -i libgstreamer*.deb
```

**Comment by bserem** [comment]
Debian also has removed libgstreamer0.10 from testing, thus TogglDesktop fails to run.
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=802812

The workaround above works, if I call TogglDesktop from the shell script.

**Comment by jmaralc** [comment]
Same issue here trying to run Scrivener:
$ scrivener
/usr/share/scrivener/bin/Scrivener: error while loading shared libraries: libgstapp-0.10.so.0: cannot open shared object file: No such file or directory

Your work around @Spurlos did not work for me :( unfortunately. Curiously if I look for the libraries:
$ ldconfig -p | grep libgstapp
	libgstapp-1.0.so.0 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libgstapp-1.0.so.0
	libgstapp-1.0.so.0 (libc6) => /usr/lib/i386-linux-gnu/libgstapp-1.0.so.0
	libgstapp-0.10.so.0 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libgstapp-0.10.so.0
	libgstapp-0.10.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libgstapp-0.10.so
Aparently they are installed.....
No idea what to do.
Thanks.

**Comment by IndrekV** [comment]
If you just installed the dependency then try to restart the terminal. It seems that app is looking in the wrong place.

For me the output is same but Toggl Desktop has no problem starting up. 
```
$ ldconfig -p | grep libgstapp
	libgstapp-1.0.so.0 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libgstapp-1.0.so.0
	libgstapp-0.10.so.0 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libgstapp-0.10.so.0
```

Probably some issue with paths in Scrievener. Please contact their support to get solution.

If it fails

**Comment by romanlevin** [comment]
The [workaround](https://github.com/toggl/toggldesktop/issues/1894#issuecomment-254427913) doesn't work for me on Ubuntu 16.10.

```shell
$ sudo dpkg -i libgstreamer-plugins-base0.10-0_0.10.36-1_amd64.deb 
Selecting previously unselected package libgstreamer-plugins-base0.10-0:amd64.
(Reading database ... 483495 files and directories currently installed.)
Preparing to unpack libgstreamer-plugins-base0.10-0_0.10.36-1_amd64.deb ...
De-configuring libgstreamer-plugins-base0.10-0:i386 (0.10.36-2) ...
Unpacking libgstreamer-plugins-base0.10-0:amd64 (0.10.36-1) ...
dpkg: error processing package libgstreamer-plugins-base0.10-0:amd64 (--install):
 package libgstreamer-plugins-base0.10-0:amd64 0.10.36-1 cannot be configured because libgstreamer-plugins-base0.10-0:i386 is at a different version (0.10.36-2)
dpkg: error processing package libgstreamer-plugins-base0.10-0:i386 (--install):
 package libgstreamer-plugins-base0.10-0:i386 0.10.36-2 cannot be configured because libgstreamer-plugins-base0.10-0:amd64 is at a different version (0.10.36-1)
Errors were encountered while processing:
 libgstreamer-plugins-base0.10-0:amd64
 libgstreamer-plugins-base0.10-0:i386
```

**Comment by Spurlos** [comment]
Look like something is off with version mismatch. Are you using Ubuntu 64bit version?

**Comment by romanlevin** [comment]
@Spurlos Yup.

```
$ uname -a
Linux laurens 4.8.0-34-generic #36-Ubuntu SMP Wed Dec 21 17:24:18 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by IndrekV** [comment]
@romanlevin 
The error suggests that you already have a version of libgstreamer installed and the version suggested in the workaround cannot be configured. Maybe try to remove the installed version and then try to install the version suggested in the workaround.

**Comment by romanlevin** [comment]
@IndrekV That's embarrassing. I was sure I did that.

Turns out I didn't! Both packages install fine now.

**Comment by Fleuv** [comment]
What is the process of this?

Do we linux users without the packages libgstreamer0.10 & libgstreamer-plugins-base0.10 have to figure out how to install those packages (downloading and installing as @joshfleming suggest is probably the most solid way of doing it). Or is there a bugfix on the way?

**Comment by IndrekV** [comment]
@Fleuv The fix will not come in near future. To fix this we would have to upgrade the application framework and this means that rewrite some parts of the application to use the newer functions. I suggest using the workaround.

**Comment by andrewhowdencom** [comment]
Also having this issue. 

FWIW, installing packages outside the canonical repositories seems like a super dirty fix. The problem is those packages by nature aren't going to receive security or stability updates, and will progressively conflict with more stuff over time (what happens when someone needs 1.0.0 for something else?)

Debian stretch is hitting stable, and doesn't package 0.10.0 (only 1.0.0) -- I think it's worth thinking about fixing this issue.

**Comment by mkurz** [comment]
Please fix this. It seems all major Linux distributions are facing that issue.

This issue was opened 1 and 1/2 years ago... And your answer to this is basically something like  "this is too much work to fix because we have to change parts of the code..." Are you serious? Even worse it looks like toggl doesn't work anymore with any (major) Linux distribution *out of the box*.

I try to get started with toggl and I am also thinking about purchasing at least the starter plan however such things scare me off. How can you wake up in the morning knowing you leave a lot of your user base behind and don't care about them? Doesn't that worry you?

**Comment by IndrekV** [comment]
@mkurz Thank you for your feedback. We have plans for a bigger upgrade of the Linux app in the first quarter of 2018 and this issue will be dealt with during this development.

**Comment by jedie** [comment]
Hope a new linux version will coming soon!

**Comment by chris-aeviator** [comment]
@IndrekV 

first quarter is almost at the end - what's the status of the Linux app so far? Running on arch and having the same problem, currently converting my not-able-to-sync client (from AUR ) Database with sqlite to csv and sort my projects manually - really not the "HELPS YOU FOCUS" experience I imagine...

Hope your solving it soon, would appreciate an update!

**Comment by IndrekV** [comment]
Hey @chris-aeviator 

Sorry for the delay with this. I already started the upgrade that would fix the old `libgstreamer` dependency issue but some plans shifted and I was forced to stop working on it. I'll try to pick it up again. Thank you for your active participation and feedback on this.

**Comment by danielkesselberg** [comment]
Hey @IndrekV do you have any updates for us?

**Comment by IndrekV** [comment]
Yes I do. We have successfully updated the app to use the newer Qt version and this fixes this dependency problem. Depending on how the testing process goes new beta version will be out sometime next week.

**Comment by Spurlos** [comment]
Yay, count me in. Will be happy to test the app in day you day usage.

**Comment by Spurlos** [comment]
I assume that beta release date have been postponed. Any new timeline on that?

**Comment by mkurz** [comment]
Yeah, any new timeline?

**Comment by IndrekV** [comment]
We decided to go on and put new improved autocomplete to the new version also and it seems that this might have been a bit too bold goal. We're still working out some issues with the new autocomplete and hope to get the new version out as soon as we are pleased with the new autocomplete.

**Comment by IndrekV** [comment]
If you are willing to test the latest version we have here is the link (this version has the update to fix this issue but does not have new autocomplete) https://github.com/toggl/toggldesktop/releases/download/v7.4.271/toggldesktop_linux_x86_64-7_4_271.tar.gz

**Comment by lee-tts** [comment]
Agreed. The old autocomplete seemed serviceable to me, but the GStreamer issue is a showstopper.

**Comment by lee-tts** [comment]
Thanks @IndrekV 
Btw is there an est. release date?

**Comment by keikoro** [comment]
Echoing what @lee-tts said, what's the ETA on this?

**Comment by bsberry** [comment]
Beta version posted in September didn't work due to some files not being marked executable. Latest update in the "beta" channel is here and works out of the box for me on a fresh 18.10 install: https://github.com/toggl/toggldesktop/releases/download/v7.4.272/toggldesktop_linux_x86_64-7_4_272.tar.gz . Not sure why it's not marked as released but seems to work fine so far.

**Comment by mcac0006** [comment]
Hi,

It looks like the dependencies are not featured on Debian 'Buster' anymore. To be able to use them, you'll need to add an older apt repository belonging to the Jessie release:

- append `deb http://ftp.de.debian.org/debian jessie main` to your `/etc/apt/sources.list`
- `apt update && apt upgrade`
- `dpkg -i toggl...deb`
- `apt --fix-broken install`
- you should be good to go!


Reference: https://packages.debian.org/jessie/amd64/libgstreamer0.10-0/download

**Comment by sitnarf** [comment]
I was able to install toggl through snap on Ubuntu 19.10:
`sudo snap install --edge toggldesktop`

**Comment by Fensterbank** [comment]
> This will be properly fixed with next release. Version 7.4.274 or later.

This issue still persists and I don't have the feeling anybody is working on this.
I think it should be reopened. The current Linux client is not usable on most modern distributions.

**Comment by MartinBriza** [comment]
@Fensterbank considering you opened an issue about a Snap package a while ago, let me point you to the fact that Snapcraft packages are still in the `edge` channel and we do not promote them anywhere officially. 
We also have a new beta release of both the tarball and Debian packages (this time for Ubuntu 18.04) which should work alright on Ubuntu derived distributions.
I am, of course, working on making it work on as many computers as possible, but there is a TON of distros and if nobody opens bugs, I can't know of any issues you can experience in, say, Budgie. So please let me know if you have any other issues, thanks.

**Comment by MartinBriza** [comment]
I'd also like to suggest you to consider trying the Flatpak package from https://flathub.org as I have much better experience (and results) with that. Compared to Snaps, Flatpaks actually are usable cross-distro.

**Comment by Fensterbank** [comment]
> @Fensterbank considering you opened an issue about a Snap package a while ago, let me point you to the fact that Snapcraft packages are still in the `edge` channel and we do not promote them anywhere officially.

Okay. I did not know that, just tried to find a way to get the app running somehow and the appearance in Gnome Software made it look officially released.
![grafik](https://user-images.githubusercontent.com/1684826/65534012-fed5e180-defe-11e9-8b98-061b8642796d.png)


> We also have a new beta release of both the tarball and Debian packages (this time for Ubuntu 18.04) which should work alright on Ubuntu derived distributions.

Thanks for this hint. I downloaded the beta (7.4.528) and can confirm it works out of the box in Ubuntu 19.04.
Ehm... I realized, I did not saw the important comments mentioning the beta releases and tried it with a stable v7.4.122. Sorry for that. 😶

**Comment by MartinBriza** [comment]
No harm done. :) But seriously, if you hit any issues, please go ahead and either contact our support (slower) or open just open a bug here on Github (usually pretty fast), we are working on all three platforms and we're trying to make our app work as smooth as possible for everyone.

**Comment by MartinBriza** [comment]
And yeah, I know Ubuntu does present the app in the Software center in this way, even though we've never done anything but `edge` channel releases. It's annoying to say the least.
